src/commands/plugins/inspect.ts

```diff
@@ -122,23 +125,11 @@ export default class PluginsInspect extends Command {
     return {...plugin, deps: depsJson} as PluginWithDeps
```

Comment:
removing this shows how much this constructed type is missing from Plugin. I think that's because of the spread operator dropping the non-public stuff from the Plugin class.

Maybe define PluginWithDeps using Pick<> or Omit<>.

dishonest/unreliable types are worse than no types.  A lot of the `as` in oclif seems to be indicators of that.